cpp/new_journal_alert.cc

```diff
@@ -172,6 +206,220 @@ namespace std {
+                     email_contents += "    </ul>\n"; // end items
+                 email_contents += "    <li>" + HtmlUtil::HtmlEscape(volume_year_and_issue) + "</li>\n";
+                 last_volume_year_and_issue = volume_year_and_issue;
+                 email_contents += "    <ul>\n"; // start items
+             }
+
+             const std::string URL("https://" + vufind_host + "/Record/" + new_issue_info.control_number_);
+             std::string authors;
+             for (const auto &author : new_issue_info.authors_)
+                 authors += "&nbsp;&nbsp;&nbsp;" + HtmlUtil::HtmlEscape(author);
+             email_contents += "      <li><a href=" + URL + ">" + HtmlUtil::HtmlEscape(new_issue_info.issue_title_) + "</a>" + authors
+                               + "</li>\n";
+         }
+         email_contents += "    </ul>\n"; // end items
+         email_contents += "  </ul>\n"; // end volume/year/issue list
+         email_contents += "</ul>\n"; // end journal list
+         email_contents += "<br />\n"
+                           "Sincerely,<br />\n"
+                           "The " + CapitalizedUserType(user_type) + " Team\n"
+                           "<br />--<br />\n"
+                           "If you have questions regarding this service please contact\n"
+                           "<a href=\"mailto:" + user_type + "@ub.uni-tuebingen.de\">" + user_type + "@ub.uni-tuebingen.de</a>.\n";
+
+         return email_contents;
+     }
+
+
+     virtual std::string generateContent(__attribute__((unused)) const std::string &vufind_host,
+                                         __attribute__((unused)) const NewIssueInfo &new_issue_infos) const
+     {
+         LOG_ERROR(std::string(__FUNCTION__) +  " currently not implemented");
+     }
+};
+
+
+class GenerateMeistertaskEmailContents :  public GenerateEmailContents {
+public:
+     virtual std::string generateContent(const std::string &vufind_host, const NewIssueInfo &new_issue_info) const
+     {
+         std::string email_contents(HtmlUtil::HtmlEscape(new_issue_info.series_title_) + " ");
+         // Generate "volume_year_and_issue":
+         std::string volume_year_and_issue;
+         if (not new_issue_info.volume_.empty())
+             volume_year_and_issue += new_issue_info.volume_;
+         if (not new_issue_info.year_.empty()) {
+             if (not volume_year_and_issue.empty())
+                 volume_year_and_issue += " ";
+             volume_year_and_issue += "(" + new_issue_info.year_ + ")";
+         }
+         if (not new_issue_info.issue_.empty()) {
+             if (not volume_year_and_issue.empty())
+                 volume_year_and_issue += ", ";
+             volume_year_and_issue += new_issue_info.issue_;
+         }
+
+         email_contents += HtmlUtil::HtmlEscape(volume_year_and_issue) + "<br/>\n";
+
+         const std::string URL("https://" + vufind_host + "/Record/" + new_issue_info.series_control_number_);
```

Comment:
Don't forget to UrlEncode the dynamic parts of the URL.

Reply:
As discussed.